config/index.js

```diff
@@ -0,0 +1,57 @@
+const fs = require('fs')
+const path = require('path')
+const flat = require('flat')
+
+/**
+ * Load an environment config for the current environment, which is controlled by
+ * process.env.HASHI_ENV
+ */
+function loadHashiConfigForEnvironment() {
+  const env = process.env.HASHI_ENV || 'development'
+  const envConfigPath = path.join(process.cwd(), 'config', `${env}.json`)
+
+  return getHashiConfig(envConfigPath)
+}
+
+/**
+ * Load an environment config from a specific path.
+ */
+function getHashiConfig(configPath) {
+  try {
+    const baseConfigPath = path.join(process.cwd(), 'config', `base.json`)
+    const baseConfig = JSON.parse(fs.readFileSync(baseConfigPath))
+    const envConfig = JSON.parse(fs.readFileSync(configPath))
+
+    // Load the extended config, if no extends property is defined use the base config
+    let extendsConfig = baseConfig
```

Comment:
If we're always going to extend from `base` when `extends` isn't set, should we omit the `extends` key from our configs?

Reply:
We could, I kinda like that it is a little more self-documenting with the explicit `extends` in each file though.

Reply:
I'm totally down to keep the `extends` in the config file. What was the reasoning behind always extending `base` even when `extends` wasn't specified? Was it more of a safety net against an author creating a new config but forgetting to extend `base`?

Reply:
@dstaley Yep, correct. The app will fail to build if all of the referenced config values aren't defined, so it seemed reasonable to default to extending base to guard against that. 

An alternative would be to validate that all keys that exist in base also exist in the final config file. I had started down this route, but it seemed potentially overkill in lieu of just the default extends.